src/main/java/server/router/routes/Redirect.java

```diff
@@ -0,0 +1,36 @@
+package server.router.routes;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import server.constants.Method;
+import server.constants.Path;
+import server.constants.Status;
+import server.request.Request;
+import server.response.Response;
+import server.response.ResponseBuilder;
+
+public class Redirect implements RouteHandler {
+  Set<Method> methodsAllowed = new LinkedHashSet<>(List.of(Method.GET));
+  Path newLocation = Path.SIMPLE_GET;
+
+  @Override
+  public Response processRequest(Request request) {
+    ResponseBuilder responseBuilder = new ResponseBuilder();
+    Method method = request.method();
+    String host = request.headers().get("Host");
+
+    switch (method) {
+      case GET, HEAD -> {
+        responseBuilder.setStartLine(Status.MOVED.format());
+        responseBuilder.addLocationHeader(host, newLocation);
+      }
+      default -> {
+        responseBuilder.setStartLine(Status.NOT_ALLOWED.format());
+        responseBuilder.addAllowHeader(methodsAllowed);
+      }
+    }
```

Comment:
I noticed that all of your routes implement this fallback in some form. We should try to define this logic in one place (`RequestRouter` looks like a good place to house this logic). If we define paths and each of their allowed methods, then we could check to see if the incoming request contains a valid path and valid method, otherwise we return the `NOT_ALLOWED` response. We can schedule some time to pair tomorrow and I can assist you on this.

Reply:
Sounds great! I'll send you an invite, thanks :)

Reply:
About to push an updated solution but happy to wait until we pair to discuss it